src/main.rs

```diff
@@ -3,12 +3,39 @@ use workspaces::Worker;
 
 #[tokio::main]
 async fn main() {
-    println!("Hello, world!");
-
     let worker = workspaces::sandbox().await.expect("should spin up sandbox");
-    profile_gas_usage(&worker, "./contracts/calculations")
+    profile_gas_usage(&worker, "./contracts/calculations", "cpu_ram_soak")
+        .await
+        .expect("should profile gas usage");
 }
 
-fn profile_gas_usage(_worker: &Worker<Sandbox>, _contract_path: &str) {
-    todo!();
+async fn profile_gas_usage(
+    worker: &Worker<Sandbox>,
+    project_path: &str,
+    method_name: &str,
+) -> anyhow::Result<()> {
+    let wasm = workspaces::compile_project(project_path).await?;
+    let contract = worker.dev_deploy(&wasm).await?;
+
+    let result = contract.call(method_name).max_gas().transact().await?;
+    let result = match result.into_result() {
+        Ok(result) => result,
+        Err(err) => return Err(anyhow::anyhow!("execution failed: {err}")),
+    };
+    let receipts = result.receipt_outcomes();
+
+    // The `FunctionCall` is the first and only action in above transaction. Only consider the gas
+    // burnt by this receipt to ignore overhead unrelated to the calculations to be benchmarked
+    // (transaction to receipt conversion, gas refunds).
+    assert_eq!(
+        2,
+        receipts.len(),
+        "transaction should generate two receipts (function call, gas refunds)",
+    );
+    let gas_burnt = receipts[0].gas_burnt;
+    println!(
```

Comment:
Just printing the gas usage is ok for debugging, but less useful for CI. I wonder if it makes sense to return the gas used as the result of this function.

Reply:
Refactored in 74dcf39. This still might not be ideal, though I think it is ok for now while trying to make the interpretation of wasm work. It can be refined later on, when it becomes more clear how benchmarks are compared and used.